### PR TITLE
feat(Create Offer): feature for user to create new offer

### PR DESCRIPTION
**Pull request by nwjbrandon**
This patch allows user to create new offer. The user needs to enter
the price and number of shares to create a new offer, a new message
pops up at the bottom.

The message is sent and received as a socket. The chatroom will be
updated according to the latest message sent.

### CODE REVIEW

File: src/reducers/ChatDux.js [hunk 1]
```diff
@@ -23,19 +22,14 @@ const chat = createSlice({
...
     addNewMessage: (state, { payload }) => {
-      const { newChat, chatRoomId } = payload;
+      const { newChat, chatRoomId, updatedAt } = payload;
       state.chatConversation.conversation.push({ ...newChat });
-      const index = _findIndex(
-        state.chatRooms,
-        c => c.chatRoomId === chatRoomId
-      );
-      state.chatRooms.splice(index, 1, payload);
+      const index = state.chatRooms.findIndex(c => c.chatRoomId === chatRoomId);
```
Comment (karrui) [inline src/reducers/ChatDux.js:27]: I thought we were changing the chat rooms to not be an array but an object with chatRoomId keys exactly so we don't have to do this findIndex thing

**Review by karrui — APPROVED** [review]
LGTM, the model probably too difficult to change now, we'll live with it. Tech debt 😂

File: src/reducers/ChatDux.js (hunk 1 above)
Comment (nwjbrandon) [reply src/reducers/ChatDux.js:27]: next PR can? If got time?Sorry

Comment (nwjbrandon) [reply src/reducers/ChatDux.js:27]: I  will put it in the issue tracker

Comment (nwjbrandon) [reply src/reducers/ChatDux.js:27]: But again when a new message comes in, we sorting according to the attribute createdAt.

Comment (karrui) [reply src/reducers/ChatDux.js:27]: Oh ya we discussed that. I think can add comment for that then haha

**Pull request merged**